### PR TITLE
Add an is_tuple_like trait for sequenceable type detection

### PR DESCRIPTION
**Pull request by Naios**
Requires #2685, otherwise the unit test won't pass

### CODE REVIEW

File: hpx/traits/is_tuple_like.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    /// has a specific tuple like size.
+    template <typename T, typename = void>
+    struct is_tuple_like
+        : std::false_type { };
+    template <typename T>
+    struct is_tuple_like<T, typename util::always_void<decltype(
+        util::tuple_size<T>::value
+    )>::type> : std::true_type { };
+}}
+
```
Comment (hkaiser) [inline hpx/traits/is_tuple_like.hpp]: Please try to adapt your style of formatting to the rest of HPX. I'll send you a .clang-format file which you can use as a starting point.

**Comment by hkaiser** [comment]
Please add the new #include here: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/include/traits.hpp

**Comment by Naios** [comment]
I fixed the code formatting and added the trait to the include compilation.
@hkaiser thank you for sending me the `.clang-format` file, it seems to be much better than the one I had.

**Comment by hkaiser** [comment]
@Naios: the current build failures are caused by a bogus commit you picked up by merging from master. Please merge again, this should resolve the issue.

**Comment by Naios** [comment]
@hkaiser I rebased the PR accordingly, but the CI is still failing

**Pull request merged**